### ISSUE TITLE
 eslint prefer static class properties

### ISSUE THREAD

**Issue by graingert** [babel/eslint-plugin-babel#128]
**Please describe what the rule should do:**
Move static class properties into the class body.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[x] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// bad
class Foo {}
Foo.bar = 3;

// good

class Foo {
  static bar = 3;
}
```

**Comment by graingert** [comment]
would also be good to disallow this. assignments that are not dependent on constructor arguments.

```js
//bad

class Foo {
    constructor(ham) {
        this.foo = 4;
    }
}

// good

class Foo {
    constructor(ham) {
        this.foo = ham;
    }
}
```

disallow class assignments.

```js
//bad

class Foo {
}

Foo.bar = 3;

// good

class Foo {
    static bar = 3;
}
```

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
